frontend/src/main/web/lib/components/SystemGlossary.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import Configs from '../constants/Configs';
+import GlossaryStore from '../stores/GlossaryStore';
+import { PureRenderMixin } from 'react/addons';
+import Actions from '../actions/GlossaryActions';
+import { Button, Input, Icons, Icon, Select } from 'zanata-ui'
+import DataTable from './glossary/DataTable'
+import NewEntryModal from './glossary/NewEntryModal'
+import ImportModal from './glossary/ImportModal'
+import { Loader } from 'zanata-ui'
+import _ from 'lodash';
+import StringUtils from '../utils/StringUtils'
+
+var SystemGlossary = React.createClass({
+  mixins: [PureRenderMixin],
+
+  _init: function() {
+    return GlossaryStore.init();
+  },
+
+  getInitialState: function() {
+    return this._init();
+  },
+
+  componentDidMount: function() {
+    GlossaryStore.addChangeListener(this._onChange);
+  },
+
+  componentWillUnmount: function() {
+    GlossaryStore.removeChangeListener(this._onChange);
+  },
+
+  _onChange: function() {
+    this.setState(this._init());
+  },
+
+  _handleTransChange: function(localeId) {
```

Comment:
Is this about changing the selected translation locale? I might need a longer name to be clear on what it does (name seems like it handles when a translation changes, but the implementation looks like it is when a locale changes).